From ce63f6e8ce6b96b4022bf0e18c8e28dc715826cf Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lbartosik@marvell.com>
Date: Thu, 25 Apr 2019 08:54:07 +0200
Subject: [PATCH 188/386] crypto: cpt - fix parsing of ucode_load string

This fixes engine group creation when ucode filename
starts with se, ie or ae prefix. In this case it was
treated as an indication of type and number of engines
to be attached instead of being treated as a fillename.

Change-Id: I510824a122269b08323d072de7d69b3f52ac84df
Signed-off-by: Lukasz Bartosik <lbartosik@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/8059
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Srujana Challa <schalla@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/crypto/cavium/cpt/common/cpt_ucode.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/crypto/cavium/cpt/common/cpt_ucode.c b/drivers/crypto/cavium/cpt/common/cpt_ucode.c
index c40fcfba215d..a501c3aed15d 100644
--- a/drivers/crypto/cavium/cpt/common/cpt_ucode.c
+++ b/drivers/crypto/cavium/cpt/common/cpt_ucode.c
@@ -1360,7 +1360,7 @@ static ssize_t ucode_load_store(struct device *dev,
 				goto err_print;
 			if (strlen(val) != 4)
 				goto err_print;
-		} else if (!strncasecmp(val, "se", 2)) {
+		} else if (!strncasecmp(val, "se", 2) && strchr(val, ':')) {
 			if (has_se || ucode_idx)
 				goto err_print;
 			tmp = strim(strsep(&val, ":"));
@@ -1372,7 +1372,7 @@ static ssize_t ucode_load_store(struct device *dev,
 				goto err_print;
 			engs[grp_idx++].type = SE_TYPES;
 			has_se = true;
-		} else if (!strncasecmp(strim(val), "ae", 2)) {
+		} else if (!strncasecmp(val, "ae", 2) && strchr(val, ':')) {
 			if (has_ae || ucode_idx)
 				goto err_print;
 			tmp = strim(strsep(&val, ":"));
@@ -1384,7 +1384,7 @@ static ssize_t ucode_load_store(struct device *dev,
 				goto err_print;
 			engs[grp_idx++].type = AE_TYPES;
 			has_ae = true;
-		} else if (!strncasecmp(val, "ie", 2)) {
+		} else if (!strncasecmp(val, "ie", 2) && strchr(val, ':')) {
 			if (has_ie || ucode_idx)
 				goto err_print;
 			tmp = strim(strsep(&val, ":"));
-- 
2.17.1

